### ISSUE TITLE
Fails to compile with gccgo

### ISSUE THREAD

**Issue by OceanS2000** [quic-go/quic-go#3168]
When I try to compile github.com/syncthing/syncthing, which uses quic-go as a dependency, with gccgo I came accross the following error:

```
% CGO_ENABLED=1 go run build.go -no-upgrade
Notice: Next generation GUI will not be built; see --with-next-gen-gui.
......
# github.com/lucas-clemente/quic-go/internal/qtls
/tmp/ccw2vu1Y.s: Assembler messages:
/tmp/ccw2vu1Y.s:5968: Error: junk `.cipherSuiteTLS13ByID ' after expression
/tmp/ccw2vu1Y.s:5968: Error: invalid operands (*UND* and *UND* sections) for `/'
/tmp/ccw2vu1Y.s:5968: Error: invalid operands (*UND* and *UND* sections) for `/'
/tmp/ccw2vu1Y.s:5968: Error: can't resolve `L0' {*ABS* section} - `go1' {*UND* section}
exit status 2
exit status 1
```

As mentioned in #2883, the problem may be with `go:linkname`. It seems to me that gccgo is looking for mangled function names here.

After manually mangling the function name (` '.'->'_0' '/'->'_1' '-'->'_x2d'`)
```go
//go:linkname cipherSuiteTLS13ByID "github_0com_1marten_x2dseemann_1qtls_x2dgo1_x2d15.cipherSuiteTLS13ByID"
```
I managed to compile the package and the resulted binary seems working fine.

Maybe a new conditional file with `// +build gccgo` can solve this better?

My `go version`:
```
go version go1.16.3 gccgo (Gentoo 11.1.0 p1) 11.1.0 linux/amd64
```

My `go env`:
```
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/<username redacted>/.cache/go-build"
GOENV="/home/<username redacted>/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/<username redacted>/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/<username redacted>/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/libexec/gcc/x86_64-pc-linux-gnu/11.1.0"
GOVCS=""
GOVERSION="go1.16.3 gccgo (Gentoo 11.1.0 p1) 11.1.0"
GCCGO="/usr/x86_64-pc-linux-gnu/gcc-bin/11.1.0/gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build577597224=/tmp/go-build -gno-record-gcc-switches -funwind-tables"
```

**Comment by marten-seemann** [comment]
> Maybe a new conditional file with // +build gccgo can solve this better?

I'd be happy to accept a PR for this.

Is there any way we can test this in CI? Maybe in the [cross-compile](https://github.com/lucas-clemente/quic-go/blob/master/.github/workflows/cross-compile.yml) test?

**Comment by OceanS2000** [comment]
> I'd be happy to accept a PR for this.

#3171
 
> Is there any way we can test this in CI? Maybe in the [cross-compile](https://github.com/lucas-clemente/quic-go/blob/master/.github/workflows/cross-compile.yml) test?

I try to use `ubuntu-latest` and install gccgo package manually as it is not included in `actions/setupgo` workflow. However gccgo in Ubuntu 20.04 seems to be too old with `go version go1.14.6 gccgo (Ubuntu 10.2.0-5ubuntu1~20.04) 10.2.0 linux/amd64`. Building a custom gccgo toolchain from source seems to be too time-consuming on each run of CI.

**Comment by marten-seemann** [comment]
I assume this has been fixed by switching to crypto/tls (for Go 1.21). Please reopen if you're still running into problems.
